RIOT-OS/RIOT
# treewide: clean up architecture specific toolchain setup

### Contribution description

Architecture specific toolchain setup is done mostly in `makefiles/arch/<ARCH>.inc.mk`, except for ESP and ARM9. Those oddballs are changed to the common pattern for consistency.

Secondly, each CPU family included the architecture specific settings on its. However, as the architecture was is already known once the features are resolved, one can simple include `$(RIOTMAKE)/arch/$(CPU_ARCH).inc.mk` directly in `$(RIOTBASE)/Makefile.include`.

### Testing procedure

If the toolchains are not properly set up, Murdock will complain. In addition, it might make sense to pick one example application and board for each platform and confirm that binaries doesn't change.

### Issues/PRs references

One step towards making https://github.com/RIOT-OS/RIOT/pull/15993 possible without reordering the processing of `Makefile.dep` and `Makefile.include`.
